srcdeps-maven-plugin/src/test/java/org/srcdeps/mvn/plugin/ExtensionsXmlUtilsTest.java

```diff
@@ -0,0 +1,177 @@
+import org.junit.Test;
+import org.w3c.dom.Document;
+
+@SuppressWarnings("deprecation")
+public class ExtensionsXmlUtilsTest {
+    private static final String EXTENSIONS_XML_TEMPLATE_REPO = //
+            "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" + //
+                    "<extensions>\n" + //
+                    "  <extension>\n" + //
+                    "    <groupId>org.srcdeps.mvn</groupId>\n" + //
+                    "    <artifactId>srcdeps-maven-local-repository</artifactId>\n" + //
+                    "    <version>%s</version>\n" + //
+                    "  </extension>\n" + //
+                    "</extensions>\n";
+
+    private static final String EXTENSIONS_XML_TEMPLATE_REPO_AND_ENFORCER = //
+            "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" + //
+                    "<extensions>\n" + //
+                    "  <extension>\n" + //
+                    "    <groupId>org.srcdeps.mvn</groupId>\n" + //
+                    "    <artifactId>srcdeps-maven-local-repository</artifactId>\n" + //
+                    "    <version>%s</version>\n" + //
+                    "  </extension>\n" + //
+                    "  <extension>\n" + //
+                    "    <groupId>org.srcdeps.mvn</groupId>\n" + //
+                    "    <artifactId>srcdeps-maven-enforcer</artifactId>\n" + //
+                    "    <version>%s</version>\n" + //
+                    "  </extension>\n" + //
+                    "</extensions>\n";
+
+    private static final String EXTENSIONS_XML_TEMPLATE_REPO_ENFORCER_AND_THIRD_PARTY = //
+            "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" + //
+                    "<extensions>\n" + //
+                    "  <extension>\n" + //
+                    "    <groupId>org.whatever</groupId>\n" + //
+                    "    <artifactId>whatever</artifactId>\n" + //
+                    "    <version>0.0.1</version>\n" + //
+                    "  </extension>\n" + //
+                    "  <extension>\n" + //
+                    "    <groupId>org.srcdeps.mvn</groupId>\n" + //
+                    "    <artifactId>srcdeps-maven-local-repository</artifactId>\n" + //
+                    "    <version>%s</version>\n" + //
+                    "  </extension>\n" + //
+                    "  <extension>\n" + //
+                    "    <groupId>org.srcdeps.mvn</groupId>\n" + //
+                    "    <artifactId>srcdeps-maven-enforcer</artifactId>\n" + //
+                    "    <version>%s</version>\n" + //
+                    "  </extension>\n" + //
+                    "</extensions>\n";
+
+    private static final String INDENT_STRING = "  ";
+
+    private static void assertUpgradeOrAdd(Document doc, String upgradeVersion, String expected) throws IOException {
+        ExtensionsXmlUtils.upgradeOrAddSrcdeps(doc, upgradeVersion);
+
+        StringWriter out = new StringWriter();
+
+        OutputFormat format = new OutputFormat(doc);
+                format.setLineWidth(128);
```

Comment:
Ah, you mean L92 should be indented by 8 spaces rather than 16? Took me long to grasp that you do not say anything about the format var and its methods.